src/HealthChecks/HealthChecks/test/HealthCheckPublisherHostedServiceTest.cs

```diff
@@ -385,6 +486,103 @@ public async Task RunAsync_CanFilterHealthChecks()
         }
     }
 
+    [Fact]
+    public async Task RunAsync_CanFilterHealthChecks_RegistrationParameters()
+    {
+        // Arrange
+        const string HealthyMessage = "Everything is A-OK";
+
+        var unblockDelayedCheck = new TaskCompletionSource<object?>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+        var publishers = new TestPublisher[]
+        {
+                new TestPublisher(),
+                new TestPublisher(),
+        };
+
+        var service = CreateService(
+            publishers,
+            configurePublisherOptions: (options) =>
+            {
+                options.Predicate = (r) => r.Name.Contains("Delay");
```

Comment:
Can we (de)select some registrations both with and without custom delays and periods in with the predicate to make extra sure we're able to filter out registrations with these customized. Maybe something like this? But having one default registration still run would be even better.

```suggestion
                options.Predicate = (r) => r.Name.Contains("Delay") && !r.Name.Contains("Delay2");
                options.Delay = TimeSpan.Zero;
```

Also, if you set `options.Delay = TimeSpan.Zero` you don't have to call `RunServiceAsync` to run the default health checks.